src/main/kotlin/org/rust/ide/inspections/lints/RsLint.kt

```diff
@@ -51,6 +51,14 @@ enum class RsLint(
                 WARN -> ProblemHighlightType.WEAK_WARNING
                 else -> super.toHighlightingType(level)
             }
+    },
+
+    UnreachablePattern("unreachable_patterns", listOf("unused")) {
+        override fun toHighlightingType(level: RsLintLevel): ProblemHighlightType =
+            when (level) {
+                WARN -> ProblemHighlightType.WARNING
```

Comment:
The default implementation does exactly the same things so you can drop `toHighlightingType` here at all